fhir/com.b2international.snowowl.fhir.rest/src/com/b2international/snowowl/fhir/rest/MetadataController.java

```diff
@@ -0,0 +1,387 @@
+import com.b2international.snowowl.fhir.core.model.dt.Uri;
+import com.b2international.snowowl.fhir.core.model.operationdefinition.OperationDefinition;
+import com.b2international.snowowl.fhir.core.model.operationdefinition.Parameter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Maps;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.extensions.Extension;
+import io.swagger.v3.oas.annotations.extensions.ExtensionProperty;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.media.Schema;
+
+/**
+ * REST end-point for serving content describing the server's capabilities.
+ * @since 8.0.0
+ */
+@Tag(description="CapabilityStatement", name = "CapabilityStatement", extensions = 
+	@Extension(name = B2I_OPENAPI_X_NAME, properties = { 
+		  @ExtensionProperty(name = B2I_OPENAPI_PROFILE, value = "http://hl7.org/fhir/StructureDefinition/CapabilityStatement")
+	}))
+@RestController
+@RequestMapping(value="/", produces = { AbstractFhirResourceController.APPLICATION_FHIR_JSON })
+public class MetadataController extends AbstractFhirResourceController<CapabilityStatement> {
+	
+	/*
+	 * Class to expose the getOpenApi() method on the superclass.
+	 */
+	@Component
+	public static class SnowOwlOpenApiWebMvcResource extends OpenApiWebMvcResource {
+		
+		@Autowired
+		public SnowOwlOpenApiWebMvcResource(ObjectFactory<OpenAPIService> openAPIBuilderObjectFactory,
+				AbstractRequestService requestBuilder, GenericResponseService responseBuilder,
+				OperationService operationParser, RequestMappingInfoHandlerMapping requestMappingHandlerMapping,
+				Optional<ActuatorProvider> actuatorProvider, Optional<List<OperationCustomizer>> operationCustomizers,
+				Optional<List<OpenApiCustomiser>> openApiCustomisers,
+				SpringDocConfigProperties springDocConfigProperties,
+				Optional<SecurityOAuth2Provider> springSecurityOAuth2Provider,
+				Optional<RouterFunctionProvider> routerFunctionProvider,
+				Optional<RepositoryRestResourceProvider> repositoryRestResourceProvider) {
+			super(openAPIBuilderObjectFactory, requestBuilder, responseBuilder, operationParser, requestMappingHandlerMapping,
+					actuatorProvider, operationCustomizers, openApiCustomisers, springDocConfigProperties,
+					springSecurityOAuth2Provider, routerFunctionProvider, repositoryRestResourceProvider);
+		}
+		
+		@Override
+		public synchronized OpenAPI getOpenApi() {
+			return super.getOpenApi();
+		}
+	}
+	
+	
+	@Autowired
+	private ObjectMapper objectMapper;
+	
+	
+	@Autowired
+	private SnowOwlOpenApiWebMvcResource openApiResource;
+	
+	//RestController is singleton scoped, only a single instances of these objects are 'cached'
```

Comment:
Yes, but multiple threads can access them at any time. Please make sure both reading and writing of these fields are thread safe.